cmd/reduce: fix logging and handle parsing errors

mgartner opened the pull request:
#### cmd/reduce: move pkg/testutils/reduce to pkg/cmd/reduce

Release note: None

#### cmd/reduce: fix logging and handle parsing errors

Previously, the `reduce` tool could unexpectedly exit without printing a
reduce form of the input SQL if a transformation pass resulted in SQL
that could not be successfully parsed. This commit ignores these errors
so that the tool can continue to reduce.

Also, the logger has been changed to a goroutine-safe `log.Logger`.

Release note: None

#### cmd/reduce: unset errexit

The `reduce` tool now unsets `errexit` in CRDB shells so that the entire
SQL input is processed, even if there are multiple errors.

Release note: None

#### cmd/reduce: simplify pretty printed SQL output

Previously, the `reduce` tool obnoxiously formatted the reduced SQL with
many parentheses, line breaks, and indents. It has been simplified so
that the output is condensed and easier to read.

Release note: None

mgartner committed:
cmd/reduce: fix logging and handle parsing errors

Previously, the `reduce` tool could unexpectedly exit without printing a
reduce form of the input SQL if a transformation pass resulted in SQL
that could not be successfully parsed. This commit ignores these errors
so that the tool can continue to reduce.

Also, the logger has been changed to a goroutine-safe `log.Logger`.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/74214)
<!-- Reviewable:end -->

RaduBerinde reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74214#-:-MrYfct9CyUIucTuBhqU:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

mgartner committed:
cmd/reduce: move pkg/testutils/reduce to pkg/cmd/reduce

Release note: None

mgartner committed:
cmd/reduce: unset errexit

The `reduce` tool now unsets `errexit` in CRDB shells so that the entire
SQL input is processed, even if there are multiple errors.

Release note: None

mgartner commented:
Had to move some things around to appease the linter. Also added a commit to set `errexit=false` in the CRDB shells spawned by `reduce`.

mgartner committed:
cmd/reduce: simplify pretty printed SQL output

Previously, the `reduce` tool obnoxiously formatted the reduced SQL with
many parentheses, line breaks, and indents. It has been simplified so
that the output is condensed and easier to read.

Release note: None

mgartner commented:
Going to go ahead and merge now that the build is passing.

TFTR!

bors r+
